pkg/ocm/helpers.go

```diff
@@ -636,6 +638,49 @@ func ValidateSubnetsCount(multiAZ bool, privateLink bool, subnetsInputCount int)
 	return nil
 }
 
+func ValidateHostedClusterSubnets(awsClient aws.Client, isPrivate bool, subnetIDs []string) (int, error) {
+	if isPrivate && len(subnetIDs) < 1 {
+		return 0, fmt.Errorf("The number of subnets for a private hosted cluster should be at least one")
+	}
+	if !isPrivate && len(subnetIDs) < 2 {
+		return 0, fmt.Errorf("The number of subnets for a public hosted cluster should be at least two")
+	}
+	vpcSubnets, vpcSubnetsErr := awsClient.GetVPCSubnets(subnetIDs[0])
```

Comment:
We get all the VPC's subnets and then filtering them to find the subnets provided by the user.

I'm wondering if we can fetch from the AWS client only the subnets with the provided IDs in the first place.
`c.ec2Client.DescribeSubnets(&ec2.DescribeSubnetsInput{SubnetIds: []*string{<subnet1>,<subnet2>...}})`

Also, it's better not to expose `[]*ec2.Subnet` AWS SDK objects outside of the client.
Can we encapsulate this functionality in the client? So we pass a list of subnets to the AWS client, and then, get a private and public count. WDYT?

I hope it doesn't require too many changes under tight timelines...

Reply:
It's a fair request; let me evaluate the effort of this.

Reply:
In other cases in the code we do sometimes need all the subnets from a VPC, for example when provisioning a day-2 machine pool and choosing a subnet from the VPC.

I didn't mean to refactor `awsClient.GetVPCSubnets`, as it's needed.

But to create a new function for this specific case, to count private/public with the provided subnets.